### ISSUE TITLE
Encoding error for dataset descriptions

### ISSUE THREAD

**Issue by PG-TUe** [openml/openml-python#126]
Hi all,

I recently ran into an issue downloading some datasets. I have been able to solve this problem, and want run my solution by you guys before making the actual change (a sanity check, if you will).

For me, running the following code would give me an error:

```
from openml import datasets
datasets.get_dataset(1515)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "openml/datasets/functions.py", line 239, in get_dataset
    description = _get_dataset_description(did)
  File "openml/datasets/functions.py", line 267, in _get_dataset_description
    fh.write(dataset_xml)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 194  
```

When first retrieving a dataset and its description, an attempt is made to write it to a local cache file. However, writing it to a file in some cases does not work. This is caused by the way in which Python 2 handles strings and unicode (so note, this is not for Python 3). 

The dataset_xml is of type `<unicode>`, and when writing it to the file it attempts to encode it with the ASCII encoder by default (at least in my case, CentOS 6.7). This goes fine as long as the string contains only ASCII letters (generally latin alphabet), but it has no support for some other characters such as é (in this case). So, we have to explicitly first encode the unicode object to a byte-array (string) with a utf-8 encoding instead, replacing `fh.write(dataset_xml)` with `fh.write(dataset_xml.encode('utf-8'))` thus solves the problem.

The `encode` function on unicode objects still works in Python 3, so as far as I can tell, adding the `.encode('utf-8')` call should not break for other versions.

**Edit:** Actually, this just moved the problem further down the line for some datasets, as liac-arff does not handle the encoding in the same way either. So the long term solution is to patch liac-arff up, but for now perhaps using `.encode("ascii","ignore")` instead of `.encode('utf-8')` might be better. It means that everything will still be using ASCII, but it comes at the price of dataloss (characters such as é will not be in the local cache).

**Furthermore:**  it seems that in the same function [_get_dataset_description](https://github.com/openml/openml-python/blob/develop/openml/datasets/functions.py#L251) the xml is written to the same file twice (the second one overwriting the first), once at line [272](https://github.com/openml/openml-python/blob/develop/openml/datasets/functions.py#L272) and then at line [284](https://github.com/openml/openml-python/blob/develop/openml/datasets/functions.py#L284). As far as I can tell the first write happens if and only if the second write also happens - the redundancy seems unnecessary. 

Link to the method for those not seeing Markdown links: https://github.com/openml/openml-python/blob/develop/openml/datasets/functions.py#L251.

[reference](https://www.azavea.com/blog/2014/03/24/solving-unicode-problems-in-python-2-7/)

**Comment by mfeurer** [comment]
Can you please check the problem was fixed by #244c585269fd824a1742103ff76b4f1b260f35bb?

**Comment by mfeurer** [comment]
I just tried reproducing this issue and failed. I will close it for now. Please reopen if necessary.
